### PR TITLE
Fix side menu icons. Add Profile dropdown for logging out on header

### PR DESCRIPTION
**Pull request by ConsM**
Fixes the sidebar menu icons and adds a profile menu dropdown.
Addresses https://tyktech.atlassian.net/jira/software/c/projects/TT/issues/TT-3310

Screenshot 2021-09-07 at 13 17 43

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (ifrim) [inline .vscode/settings.json:1]: I think this file was committed by mistake. You probably need to add `.vscode` in `.gitignore`.

File: views/assets/javascripts/tyk.js [hunk 2]
```diff
@@ -11,3 +11,13 @@ function onCancel() {
 	}
 }
 
+document.getElementById("profile-icon").addEventListener("mouseover", mouseOver);
+document.getElementById("logout-btn").addEventListener("mouseout", mouseOut);
```
Comment (ifrim) [inline views/assets/javascripts/tyk.js:15]: Should these two events be attached to different elements? It seems weird.

File: views/assets/stylesheets/components/header.css [hunk 3]
```diff
@@ -105,6 +101,33 @@ a.qor-slideout__fullscreen > i.tyk-icon+.tyk-icon {
...
+  top: 45px;
+  right: 45px;
+  z-index: 10;
+  visibility: hidden;
+}
+
+.profile-details-container__list > li {
+  padding-block: 10px;
+  padding-left: 20px;
+  padding-right: 50px;
```
Comment (ifrim) [inline views/assets/stylesheets/components/header.css:118]: I don't think it is a good idea to mix the use of logical padding and physical padding. Unless this is intentional I would change this to use either only logical padding css properties, or only physical padding ones.

**Review by ifrim — APPROVED** [review]

File: views/assets/javascripts/tyk.js (hunk 2 above)
Comment (ConsM) [reply views/assets/javascripts/tyk.js]: Can you check again please, just modified it slightly.
The idea is to show the "dropdown" menu when hovering over the icon but hide it when hovering elsewhere

**Comment by lghiur** [comment]
@ConsM what is the Jira issue number? can we see some designs to compare with?

File: views/assets/javascripts/tyk.js (hunk 2 above)
Comment (lghiur) [reply views/assets/javascripts/tyk.js]: @ConsM you can do this only from css, no js needed. 
```
.profile-icon:hover + profile-menu {
    display: block;
}
```

File: views/assets/javascripts/tyk.js [hunk 4]
```diff
@@ -11,3 +11,14 @@ function onCancel() {
 	}
 }
 
+document.getElementById("profile-icon").addEventListener("mouseover", showProfileMenu);
```
Comment (lghiur) [inline views/assets/javascripts/tyk.js:14]: @ConsM the code in this file might not be needed and a pure CSS solution might work. Even so, if you get to write code like this, you could try to organised it better even if it's a small piece of functionality

Here's how I would've done it

```
//profile-menu.js
function ProfileMenu ({id}) {
  const $element = document.getElementById(id);

  const toggleMenu = (action) => {
    $element.style.visibility = action;
  }

  const show = () => {
    toggleMenu("visible");
  }

  const hide = () => {
    toggleMenu("hidden");
  }

  return {
    show,
    hide
  }
};

export default ProfileMenu;

//tyk.js

import ProfileMenu from './profile-menu';

const profileMenu = ProfileMenu({id: "profile-menu"});

document.getElementById("profile-icon").addEventListener("mouseover", profileMenu.show);
document.getElementById("logout-btn").addEventListener("mouseout", profileMenu.hide);
```

File: views/assets/javascripts/tyk.js (hunk 2 above)
Comment (ConsM) [reply views/assets/javascripts/tyk.js]: @lghiur with this approach when the cursor moves out of the icon the menu disappears instantly , its impossible to hit the logout btn

Comment (lghiur) [reply views/assets/javascripts/tyk.js]: @ConsM can we try smth like this? https://www.w3schools.com/css/css_dropdowns.asp

Comment (lghiur) [reply views/assets/javascripts/tyk.js]: if you make icon wrapper sticked to the dropdown one, it shouldn't dissapear

Comment (lghiur) [reply views/assets/javascripts/tyk.js]: https://www.w3schools.com/css/tryit.asp?filename=trycss_dropdown_text

**Comment by ConsM** [comment]
> @ConsM what is the Jira issue number? can we see some designs to compare with?

no jira issue, it was mentioned in a thread and I included it in this PR. But now that you said , it might worth adding one. Will create and link it

File: views/assets/stylesheets/components/header.css [hunk 5]
```diff
@@ -152,6 +135,25 @@ a.qor-slideout__fullscreen > i.tyk-icon+.tyk-icon {
 	}
 }
 
+@media only screen and (min-width: 320px) and (max-width: 500px) {
```
Comment (lghiur) [inline views/assets/stylesheets/components/header.css:138]: @ConsM  indentation looks off

File: views/assets/javascripts/modules/profile-menu.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+export function ProfileMenu ({id}) {
```
Comment (lghiur) [inline views/assets/javascripts/modules/profile-menu.js:1]: @ConsM have you tried the css aproach?

Comment (ConsM) [reply views/assets/javascripts/modules/profile-menu.js]: just did it, works fine indeed. Thanks

File: views/assets/stylesheets/components/header.css (hunk 5 above)
Comment (ConsM) [reply views/assets/stylesheets/components/header.css:147]: formatted the whole document

**Review by padiazg — APPROVED** [review]
LGTM

**Comment by ConsM** [comment]
@lghiur can we merge now ?

**Comment by lghiur** [comment]
@ConsM please apply same formatting to the files as on Dashboard or ARA, we can merge after that

**Pull request merged**